x/distribution/spec/03_begin_block.md

```diff
@@ -0,0 +1,80 @@
+<!--
+order: 3
+-->
+
+# Begin Block
+
+At each `BeginBlock`, the fees received in the previous block are transferred to the distribution `ModuleAccount`, as it's the account the one who keeps track of the flow of coins in (as in this case) and out the module. The fees are also allocated to the proposer, community fund and global pool. When the validator is the proposer of the round, that validator (and their delegators) receives between 1% and 5% of fee rewards, the reserve community tax is then charged, then the remainder is distributed proportionally by voting power to all bonded validators independent of whether they voted (social distribution). Note the social distribution is applied to proposer validator in addition to the proposer reward.
+
+The amount of proposer reward is calculated from pre-commits Tendermint messages in order to incentivize validators to wait and include additional pre-commits in the block. All provision rewards are added to a provision reward pool which validator holds individually (`ValidatorDistribution.ProvisionsRewardPool`).
+
+## The Distribution Scheme
+
+See [params](07_params.md) for description of parameters.
+
+Let `fees` be the total fees collected in the previous block. All fees are
+collected in a specific module account during the block. During `BeginBlock`,
+they are sent to the `"distribution"` `ModuleAccount`. No other sending of
+tokens occur. Instead, the rewards each account is entitled to are stored, and
+withdrawals can be triggered through the messages `FundCommunityPool`,
+`WithdrawValidatorCommission` and `WithdrawDelegatorReward`.
+
+### Reward to the Community Pool
+
+The community pool gets `communitytax * fees`, plus any remaining dust after
+validators get their rewards, which will always be rounded down to nearest
+integer value.
+
+### Reward To the Validators
+
+The proposer will receive a base reward of `fees * baseproposerreward`. In
+addition, they will receive a bonus of `fees * bonusproposerreward * P`, where
+`P = (precommits included / total bonded validator power)`. The more precommits
```

Comment:
let's be more precise here. On the right hand of the equation we have different units: `total bonded validator power` is measured in staked tokens,  `precommits_included` is measured in number of precommits. I think the precommits must be also scaled based on the bond of the validator who sent the precommit.

Reply:
Also `total bonded validator power` might be ambiguous: 
* sum of all bonds of all validators (`sum bonds`)
* single validator total bond.

That's why I suggest to use `sum bonds`

Reply:
I'm not so sure about the term `bonds`, it doesn't really exist anywhere. There is a certain number of validators which are "bonded", meaning they are participating in consensus. But that doesn't mean there is a `bond` in any sense.

You're right about the units. I was hoping that "precommits" would be understandable in context as "the power represented by all included precommits", but clearly that's not the case.

Reply:
BTW, thte `bonusproposerreward` word should have use proper word separators.